hedera-node/src/test/java/com/hedera/services/txns/schedule/ScheduleDeleteTransitionLogicTest.java

```diff
@@ -39,12 +45,74 @@ private void setup() {
         store = mock(ScheduleStore.class);
         ledger = mock(HederaLedger.class);
         accessor = mock(PlatformTxnAccessor.class);
-
         txnCtx = mock(TransactionContext.class);
-
         subject = new ScheduleDeleteTransitionLogic(validator, store, ledger, txnCtx);
     }
 
+    @Test
+    public void followsHappyPath() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.delete(schedule)).willReturn(OK);
+
+        // when:
+        subject.doStateTransition();
+
+        // then
+        verify(store).delete(schedule);
+        verify(txnCtx).setStatus(SUCCESS);
+    }
+
+    @Test
+    public void capturesInvalidScheduleId() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.delete(schedule)).willReturn(INVALID_SCHEDULE_ID);
+
+        // when:
+        subject.doStateTransition();
+
+        // then
+        verify(store).delete(schedule);
+        verify(txnCtx).setStatus(INVALID_SCHEDULE_ID);
+    }
+
+    @Test
+    public void capturesImmutableSchedule() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.delete(schedule)).willReturn(SCHEDULE_IS_IMMUTABLE);
+
+        // when:
+        subject.doStateTransition();
+
+        // then
+        verify(store).delete(schedule);
+        verify(txnCtx).setStatus(SCHEDULE_IS_IMMUTABLE);
+    }
+
+    @Test
+    public void capturesAlreadyDeletedSchedule() {
+        // given:
+        givenValidTxnCtx();
+
+        // and:
+        given(store.delete(schedule)).willReturn(SCHEDULE_WAS_DELETED);
+
+        // when:
+        subject.doStateTransition();
+
+        // then
+        verify(store).delete(schedule);
+        verify(txnCtx).setStatus(SCHEDULE_WAS_DELETED);
+    }
```

Comment:
nit: Although those tests are perfectly fine, they are testing the inner logic of the store. Just as information and feedback -> if you want to test the schedule delete transition logic you should test that:
"if the store fails for some reason with a response code, the transition logic should fail as-well and set the appropriate status" This means that if you have only this test:

```Java
@Test
    public void capturesInvalidSchedule() {
        // given:
        givenValidTxnCtx();

        // and:
        given(store.delete(schedule)).willReturn(NOT_OK);

        // when:
        subject.doStateTransition();

        // then
        verify(store).delete(schedule);
        verify(txnCtx).setStatus(NOT_OK);
    }
```

it should be fine and you are covering those use-cases

Reply:
`capturesInvalidScheduleId`, `capturesImmutableSchedule` and `capturesAlreadyDeletedSchedule` are redundant and can be tested with only one test